### DISCUSSION TITLE
fill image with color by alpha channel

### DISCUSSION THREAD

**Discussion by Tosyk** [ImageMagick/ImageMagick#4288] [Digital Image Processing] [answered]
hi, I have an image with a pink color in it. This image has alpha channel.
I need to fill the pink color of this image with black color but only those pixels that are transparent in alpha channel

here's the image: [hnasn_7_n.zip](https://github.com/ImageMagick/ImageMagick/files/7249723/hnasn_7_n.zip)
here's preview:
![image](https://user-images.githubusercontent.com/24209966/135222611-27130d51-c796-4ea7-a7f4-0673fc27c5f3.png)

**Comment by snibgo** [comment]
I think you simply want to flatten against a black background:
```
magick hnasn_7_n.TGA -background Black -layers Flatten out.png
```

**Reply by Tosyk** [reply]
thanks, this seem to be working, but how can I keep the alpha channel?

**Comment by snibgo** [comment, marked as answer]
I'm not sure what you want. Perhaps you want to make all pixels that are fully transparent into transparent black:
```
magick hnasn_7_n.TGA -background Black -alpha background out.png
```

**Reply by Tosyk** [reply]
thank you very much, exactly what I wanted!
